source/common/http/http1/codec_impl.cc

```diff
@@ -449,11 +478,17 @@ size_t ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
 }
 
 void ConnectionImpl::onHeaderField(const char* data, size_t length) {
+  ENVOY_CONN_LOG(trace, "onHeaderField: data={}", connection_, absl::string_view(data, length));
+  // We previously already finished up the headers, these headers are
+  // now trailers
   if (header_parsing_state_ == HeaderParsingState::Done) {
-    // Ignore trailers.
-    return;
+    if (!enable_trailers_) {
+      // Ignore trailers.
+      return;
+    }
+    processing_trailers_ = true;
+    header_parsing_state_ = HeaderParsingState::Field;
```

Comment:
Please unit test pipelined requests with trailers just so we verify the http_parser state machine works well.

Reply:
What do you mean by pipelined requests?

Reply:
Ohh yea that's a good callout. It might break. I'll add a test for this.